Keep all particles in [0,L) and carefully check periodic conditions

PalPaulD opened the pull request:
Enforce all particles to lie in [0,L) in both abacus_base.py and in abacus_small.py. check_catalog() ensures that the box is safe

epaillas commented on scripts/emc/measurements/abacus_base.py:
```diff
@@ -50,6 +51,11 @@ def get_hod_positions(filename, los='z'):
         boxsize = np.array([2000/qperp, 2000/qperp, 2000/qpar])
+
+    # Make sure the catalog has all galaxies are inside expected ranges
+    pos += boxsize / 2
```
Can we do the check without modifying the positions? Almost all statistics in this script assume `boxcenter = 0`

epaillas commented on scripts/emc/measurements/abacus_base.py:
```diff
@@ -300,7 +306,7 @@ def compute_dd_knn(output_fn, positions, boxsize, los='z', **attrs):
     assert np.all(positions[:,0].astype(np.float32) < np.float32(boxsize[0])), f'{repr(np.max(positions[:,0]))} falls out of the box with size {repr(boxsize[0])} along the z-axis'
     assert np.all(positions[:,1].astype(np.float32) < np.float32(boxsize[1])), f'{repr(np.max(positions[:,1]))} falls out of the box with size {repr(boxsize[1])} along the z-axis'
-    assert np.all(positions[:,2].astype(np.float32) < np.flot32(boxsize[2])), f'{repr(np.max(positions[:,2]))} falls out of the box with size {repr(boxsize[2])} along the z-axis'
+    assert np.all(positions[:,2].astype(np.float32) < np.float32(boxsize[2])), f'{repr(np.max(positions[:,2]))} falls out of the box with size {repr(boxsize[2])} along the z-axis'
```
Now that you've implemented a dedicated function for this in `utils`, can we outsource all of this checks as calls to those functions to keep the measurements script less cluttered?

epaillas commented:
The `check_box` branch is a few commits behind cosmodesi. There will be conflicts with the [-L/2, L/2] v/s [0, L] convention and we need to pay attention to those.

PalPaulD commented:
Moved everything to [-L/2,L/2) and cleaned up my code to avoid double checking. Also almost all merging conflicts were about unrelated pieces of codes. Those that weren't had a call for checking function.

epaillas commented on scripts/emc/measurements/abacus_base.py:
```diff
@@ -573,3 +586,17 @@ def compute_dd_knn(output_fn, positions, boxsize, los='z', **attrs):
                         hod_positions, boxsize = get_hod_positions(hod_fn, los='z')
                         compute_dd_knn(output_fn, hod_positions, boxsize, los='z')
+                        box_args = dict(boxsize=boxsize, boxcenter=0.0)
+                        compute_spherical_voids(output_fn, hod_positions, **box_args)
```
Yeah, it should have highlighted the relevant lines, not sure what happened
